src/UserService.Business/Commands/User/EditUserCommand.cs

```diff
@@ -25,334 +28,356 @@
+        IRequestClient<IChangeUserRoleRequest> rcRole,
+        IRequestClient<IChangeUserOfficeRequest> rcOffice,
+        IHttpContextAccessor httpContextAccessor,
+        IBus bus)
+    {
+      _userRepository = userRepository;
+      _credentialsRepository = credentialsRepository;
+      _mapperUser = mapperUser;
+      _avatarRepository = avatarRepository;
+      _userAvatarMapper = userAvatarMapper;
+      _createImageDataMapper = createImageDataMapper;
+      _accessValidator = accessValidator;
+      _logger = logger;
+      _rcImage = rcImage;
+      _rcDepartment = rcDepartment;
+      _rcPosition = rcPosition;
+      _rcRole = rcRole;
+      _rcOffice = rcOffice;
+      _httpContextAccessor = httpContextAccessor;
+      _bus = bus;
+    }
+
+    /// <inheritdoc/>
+    public OperationResultResponse<bool> Execute(Guid userId, JsonPatchDocument<EditUserRequest> patch)
+    {
+      Operation<EditUserRequest> positionOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.PositionId), StringComparison.OrdinalIgnoreCase));
+      Operation<EditUserRequest> departmentOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.DepartmentId), StringComparison.OrdinalIgnoreCase));
+      Operation<EditUserRequest> roleOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.RoleId), StringComparison.OrdinalIgnoreCase));
+      Operation<EditUserRequest> officeOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.OfficeId), StringComparison.OrdinalIgnoreCase));
+      Operation<EditUserRequest> isActiveOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.IsActive), StringComparison.OrdinalIgnoreCase));
+
+      Guid requestSenderId = _httpContextAccessor.HttpContext.GetUserId();
+
+      if (!(_userRepository.Get(_httpContextAccessor.HttpContext.GetUserId()).IsAdmin ||
+          _accessValidator.HasRights(Rights.AddEditRemoveUsers) ||
+          (userId == requestSenderId
+          && patch.Operations.FirstOrDefault(o => o.path.EndsWith(nameof(EditUserRequest.Rate), StringComparison.OrdinalIgnoreCase)) == null
+          && positionOperation == null
+          && departmentOperation == null
+          && roleOperation == null
+          && officeOperation == null)))
+      {
+        throw new ForbiddenException("Not enough rights.");
+      }
+
+      List<string> errors = new List<string>();
+
+      Operation<EditUserRequest> imageOperation = patch.Operations.FirstOrDefault(
+          o => o.path.EndsWith(nameof(EditUserRequest.AvatarImage), StringComparison.OrdinalIgnoreCase));
+      Guid? imageId = null;
+
+      if (imageOperation != null
+        && (imageOperation.OperationType == OperationType.Add
+          || imageOperation.OperationType == OperationType.Replace))
+      {
+        imageId = GetAvatarImageId(
```

Comment:
you add images across the image controller. You can only change the current avatar id here